chaoscenter/authentication/api/handlers/rest/user_handlers.go

```diff
@@ -0,0 +1,346 @@
+		err = service.CheckPasswordHash(user.Password, userRequest.Password)
+		if err != nil {
+			log.Warn(err)
+			c.JSON(utils.ErrorStatusCodes[utils.ErrInvalidCredentials], presenter.CreateErrorResponse(utils.ErrInvalidCredentials))
+			return
+		}
+
+		token, err := user.GetSignedJWT()
+		if err != nil {
+			log.Error(err)
+			c.JSON(utils.ErrorStatusCodes[utils.ErrServerError], presenter.CreateErrorResponse(utils.ErrServerError))
+			return
+		}
+		expiryTime := time.Duration(utils.JWTExpiryDuration) * 60
+
+		var defaultProject string
+		ownerProjects, err := service.GetOwnerProjectIDs(c, user.ID)
+
+		if len(ownerProjects) > 0 {
+			defaultProject = ownerProjects[0]
+		} else {
+			// Adding user as project owner in project's member list
+			newMember := &entities.Member{
+				UserID:     user.ID,
+				Role:       entities.RoleOwner,
+				Invitation: entities.AcceptedInvitation,
+				JoinedAt:   strconv.FormatInt(time.Now().Unix(), 10),
+			}
+			var members []*entities.Member
+			members = append(members, newMember)
+			state := "active"
+			newProject := &entities.Project{
+				ID:      uuid.Must(uuid.NewRandom()).String(),
+				Name:    user.UserName + "'s project",
+				Members: members,
+				State:   &state,
+				Audit: entities.Audit{
+					IsRemoved: false,
+					CreatedAt: time.Now().Unix(),
+					CreatedBy: user.ID,
+					UpdatedAt: time.Now().Unix(),
+					UpdatedBy: user.ID,
+				},
+			}
+			err := service.CreateProject(newProject)
+			if err != nil {
+				return
+			}
+			defaultProject = newProject.ID
+		}
+
+		c.JSON(200, gin.H{
+			"access_token": token,
+			"project_id":   defaultProject,
+			"project_role": entities.RoleOwner,
+			"expires_in":   expiryTime,
+			"type":         "Bearer",
+		})
+	}
+}
+
+func UpdatePassword(service services.ApplicationService) gin.HandlerFunc {
+	return func(c *gin.Context) {
+		var userPasswordRequest entities.UserPassword
+		err := c.BindJSON(&userPasswordRequest)
+		if err != nil {
+			log.Warn(err)
+			c.JSON(utils.ErrorStatusCodes[utils.ErrInvalidRequest], presenter.CreateErrorResponse(utils.ErrInvalidRequest))
+			return
+		}
+		username := c.MustGet("username").(string)
+		userPasswordRequest.Username = username
+		if utils.StrictPasswordPolicy {
+			err := utils.ValidateStrictPassword(userPasswordRequest.NewPassword)
+			if err != nil {
+				c.JSON(utils.ErrorStatusCodes[utils.ErrStrictPasswordPolicyViolation], presenter.CreateErrorResponse(utils.ErrStrictPasswordPolicyViolation))
+				return
+			}
+		}
+		if userPasswordRequest.NewPassword == "" {
+			c.JSON(utils.ErrorStatusCodes[utils.ErrInvalidRequest], presenter.CreateErrorResponse(utils.ErrInvalidRequest))
+			return
+		}
+		err = service.UpdatePassword(&userPasswordRequest, true)
+		if err != nil {
+			log.Info(err)
+			c.JSON(utils.ErrorStatusCodes[utils.ErrInvalidCredentials], presenter.CreateErrorResponse(utils.ErrInvalidCredentials))
+			return
+		}
+		c.JSON(200, gin.H{
+			"message": "password has been reset",
```

Comment:
```suggestion
			"message": "password has been updated successfully",
```
We have different handler for reset below this function